packages/happychat-connection/src/connection.ts

```diff
@@ -1,33 +1,39 @@
 import debugFactory from 'debug';
 import IO from 'socket.io-client';
-import {
-	receiveAccept,
-	receiveConnect,
-	receiveDisconnect,
-	receiveError,
-	receiveInit,
-	receiveLocalizedSupport,
-	receiveMessage,
-	receiveMessageOptimistic,
-	receiveMessageUpdate,
-	receiveReconnecting,
-	receiveStatus,
-	receiveToken,
-	receiveUnauthorized,
-	requestTranscript,
-} from 'calypso/state/happychat/connection/actions';
+import type { Action, HappychatUser, ConnectionProps, HappychatAuth, Socket } from './types';
 
 const debug = debugFactory( 'calypso:happychat:connection' );
 
-const buildConnection = ( socket ) =>
+const buildConnection: ( socket: string | Socket ) => Socket = ( socket ) =>
 	typeof socket === 'string'
 		? new IO( socket, {
 				// force websocket connection since we no longer have sticky connections server side.
 				transports: [ 'websocket' ],
 		  } ) // If socket is an URL, connect to server.
 		: socket; // If socket is not an url, use it directly. Useful for testing.
 
-class Connection {
+export class Connection {
+	receiveAccept?: ( accept: boolean ) => void;
+	receiveConnect?: () => void;
+	receiveDisconnect?: ( reason: string ) => void;
+	receiveError?: ( error: string ) => void;
+	receiveInit?: ( init: HappychatUser ) => void;
+	receiveLocalizedSupport?: ( accept: boolean ) => void;
+	receiveMessage?: ( message: string ) => void;
+	receiveMessageOptimistic?: ( message: string ) => void;
+	receiveMessageUpdate?: ( message: string ) => void;
+	receiveReconnecting?: () => void;
+	receiveStatus?: ( status: string ) => void;
+	receiveToken?: () => void;
+	receiveUnauthorized?: ( message: string ) => void;
+	requestTranscript?: () => void;
+	dispatch: any;
+	openSocket: any;
```

Comment:
I'm not sure about this suggestion, but here goes: 

```suggestion
	dispatch: (arg: Action) => void;
	openSocket: Socket | undefined;
```